src/components/toast/index.tsx

```diff
@@ -13,8 +20,10 @@ function toast({
   description,
   icon,
   title,
-}: Omit<ToastDisplayProps, 'onDismiss'>) {
-  return reactHotToast((t: Toast) => {
+  autoDismiss = true,
+}: Omit<ToastDisplayProps, 'onDismiss'> & toastOptions) {
+  const toastFn = autoDismiss ? reactHotToast : reactHotToast.loading
```

Comment:
there also is the ability to pass `duration: Infinity` as a toast option to prevent it from auto dismissing. https://react-hot-toast.com/docs/toast#default-durations

Not sure if one way or the other is better. Went the duration route on the new consent manager setup so far. https://github.com/hashicorp/react-components/pull/590/files#diff-675b091c60a57fc2d30f112facaabff26bf5ddeaa0c067a5cbcccf01c6e4fa52R109

Reply:
@alexcarpenter Since you're here, any thoughts on `role="dialog"` and related dialog behaviour for toasts with interactive content? We're running into a related conundrum here, since we may actually want this "install command copied" toast to have a link to a getting started guide 🤔

Reply:
Not sure how that'd get announced to the user since currently `role="status"` gets applied so that when it's rendered, the contents are read to the user. If that was updated to use `role="dialog"` I don't think it would be read to the user. Not sure the recommendation for interactive content within a toast outside of maybe not auto dismissing it?

https://user-images.githubusercontent.com/825855/169566127-f84ecbb1-86eb-4988-8972-ccde2c0eb6d4.mp4

Reply:
Dialog role would interrupt the users flow vs role status does not effect flow. Since with a dialog, you'd move focus into the dialog on show.

Reply:
I think the concern with interactive toasts with `role="status"` is that the interactive content will be announced, but the interactivity will be difficult to access through keyboard navigation.

For example, imagine there's some button or action that, after receiving the toast notification, I want to activate. With the dismiss button I guess 🤷‍♂️ cause the notification will be auto-dismissed soon, and assistive tech users will receive the notification and the "Dismiss" action and hopefully ignore it. But imagine it's a critical action, or even just a link or "accept consent" button that I receive, and then want to get to and activate. How do I do that?

![CleanShot 2022-05-20 at 13 12 46](https://user-images.githubusercontent.com/4624598/169579558-75578592-7df7-4186-a603-a9c2905ff5ee.gif)

On the react-hot-toast website, the toasts are rendered in the DOM before the footer, it seems. So it seems I'd have to furiously tab to the end of the page to interact with toast content. And once I'm done interacting with the toast, I'd have to tab back to my original position. Seems like a lot of tabbing!

With that in mind, it seems like interactive content in toasts maybe does merit `role="dialog"` and interruption 🤔 Otherwise, it seems really difficult to surface the interactive content to keyboard users.

Does that track? Definitely thinking out loud here, very open to input on this 🙇‍♂️

Reply:
Yeah, I am tracking with your logic here. I am thinking it's not a one size fits all solution for this. Because I wouldn't assume clicking copy to clipboard button would auto focus me into the toast as a dialog even if there was a link in there to learn more within it. I am wondering if there is some alternative method to move to the toast via screen reader that we might not be aware of?

Reply:
☝️+1 on "alternative method". My immediate thought was "well react-hot-toast said it's accessible, so i must be doing something wrong!" 🤣 

This is a copy of a similar msg in dev-dot-engineers, but I think there are broadly four use cases:

1. ✅ Interruptive, text-only toast - `role="alert"`. `react-hot-toast` does this fine.
2. ✅ Non-urgent, text-only toast - `role="status"`. `react-hot-toast` does this fine too.
3. ✅ ( ❓)  Interruptive, interactive "toast" - `role="dialog"`. `react-hot-toast` fails here on its own... but I think we can hack around a bit to basically make a dialog that looks like a toast and renders in the bottom corner
    - Have started on this here in this PR! Basically hacking around Reach dialog, to prevent it from rendering into a portal, but otherwise using it. Definitely very broken right now (I need to figure out focus management, and remove duplicate dialog roles), but maybe a useful path.
4. ❓ Non-interruptive, interactive "toast" - it feels unclear how to do this, there doesn't seem to be much blessed accessibility guidelines on this. But we clearly have use cases - consent-manager seems like one? Maybe my problem is I've been searching for like "interactive toast patterns", and instead should search for "non-interruptive dialog pattern" 🤔